File: src/gfx/texture.cpp
/*
 * Copyright (C) 2014 Luke San Antonio
 * All rights reserved.
 */
#include "texture.h"
namespace redc
{
  void Texture::allocate(Vec<int> const& extents, Texture_Format form,
                         Texture_Target type) noexcept
  {
    extents_ = extents;
    allocate_(extents, form, type);
  }
}

File: src/lua/texture.lua

-- Copyright (c) 2016 Luke San Antonio
-- All rights reserved.

local ffi = require("ffi")

local rc = require("redcrane")

local texture = {}

function texture:load_texture(str)
   return ffi.gc(ffi.redc_load_texture(rc.engine, str),
                 ffi.C.redc_unload_texture)
end

return texture
